packages/rpc-types/src/lamports.ts

```diff
@@ -46,18 +49,45 @@ export function lamports(putativeLamports: bigint): LamportsUnsafeBeyond2Pow53Mi
     return putativeLamports;
 }
 
+type ExtractAdditionalProps<T, U> = Omit<T, keyof U>;
+
 export function getLamportsEncoder(): FixedSizeEncoder<LamportsUnsafeBeyond2Pow53Minus1, 8> {
     return getMemoizedU64Encoder();
 }
 
+export function getLamportsEncoderFromEncoder<TEncoder extends NumberEncoder>(
+    innerEncoder: TEncoder,
+): Encoder<LamportsUnsafeBeyond2Pow53Minus1> & ExtractAdditionalProps<TEncoder, NumberEncoder> {
+    return innerEncoder;
+}
+
 export function getLamportsDecoder(): FixedSizeDecoder<LamportsUnsafeBeyond2Pow53Minus1, 8> {
     return transformDecoder(getMemoizedU64Decoder(), lamports);
```

Comment:
Here I would delegate to the other helper function that accepts an inner encoder/decoder. That way if we need to change its behaviour in the future, we have a single point of failure.

Reply:
Good idea, updated :)